### ISSUE TITLE
AutoComplete: screen readers output the last menu item "MoreText" as undefined.

### ISSUE THREAD

**Issue by prusuteodora** [primefaces/primefaces#14183]
### Describe the bug

In the autoComplete component, the last item in the menu that indicates that there are more items is rendered as undefined by the screen readers.

### Reproducer

Go to any autoComplete component that has a maxResult attribute set to a value and that has the moreText attribute to have a value, and when read with the screen reader it will be rendered as undefined. There is also an example in the primefaces showcase.

<img width="1892" height="862" alt="Image" src="https://github.com/user-attachments/assets/363a6c5c-a65b-4b58-b3ce-11d46f5917f3" />

### Expected behavior

The moreText attribute value is read by the screen reader inside the autocomplete component that has an maxResults attribute with it's value set.(ex. by adding aria-label)

### PrimeFaces edition

None

### PrimeFaces version

14.0.12

### JSF implementation

Mojarra

### JSF version

4.0

### Java version

17.0.9

**Comment by prusuteodora** [comment]
The fix does not open the whole problem. t also needs added the data-item-label in order to be read when navigating with the arrow keys. The provided fix makes the moreText red by the screen reader only when hovering over it.

**Comment by melloware** [comment]
@prusuteodora from looking at all the other list items I think it ONLY needs data-item-label and not aria-label as all the other options don't have aria-label?

**Comment by melloware** [comment]
also I am starting to  think this is a bug with NVDA.  Why isn't it just reading the value of the `<li>`

**Comment by melloware** [comment]
@prusuteodora can you open a bug ticket on NVDA: https://github.com/nvaccess/nvda/issues

there is absolutely no reason it should not be reading the More Text of the `<li>Hello My Name Is<li>` should read "Hello My Name Is".

**Comment by prusuteodora** [comment]
> [@prusuteodora](https://github.com/prusuteodora) from looking at all the other list items I think it ONLY needs data-item-label and not aria-label as all the other options don't have aria-label?

That is good, from my p.o.v. Thank you for the fix!

**Comment by melloware** [comment]
@prusuteodora i am not going to fix this yet i think you should report a bug to NVDA: https://github.com/nvaccess/nvda/issues  there is nothing wrong with our code i think their reader is wrong.

**Comment by prusuteodora** [comment]
> [@prusuteodora](https://github.com/prusuteodora) i am not going to fix this yet i think you should report a bug to NVDA: https://github.com/nvaccess/nvda/issues there is nothing wrong with our code i think their reader is wrong.

I added a ticket. https://github.com/nvaccess/nvda/issues/19006

**Comment by melloware** [comment]
Thank you i just upvoted your ticket and will watch it.

**Comment by prusuteodora** [comment]
https://github.com/nvaccess/nvda/issues/19006#issuecomment-3349353383

**Comment by melloware** [comment]
@prusuteodora got to the bottom of the issue!

**Comment by prusuteodora** [comment]
okay. Thank you! :)

**Comment by melloware** [comment]
@prusuteodora this is definitely in 15.0.9 and higher are you saying you are not seeing it?  The showcase has 15.0.13 running right now can you check?

**Comment by prusuteodora** [comment]
I checked in the showcase(15.0.13) and it is still undefined: 

<img width="1256" height="890" alt="Image" src="https://github.com/user-attachments/assets/aa7fae18-76f1-4d6b-8c27-66948d93528f" />

**Comment by melloware** [comment]
interesting

**Comment by melloware** [comment]
@prusuteodora it had an `aria-label` before and now its removed to just have this HTML

```xml
<li id="j_id_47:acMaxResults_moretext" 
class="ui-autocomplete-item ui-autocomplete-moretext" 
role="option" aria-selected="false">more items available</li>
```
Your screen reader should be reading "more items available".

**Comment by melloware** [comment]
i don't think `undefined` is related as you can see in your screen shot your screenreader has `more items available 6 of 6`

**Comment by melloware** [comment]
i think NVDA was saying `the real fix was simply the MORE TExt needed an ID field so aria-activedescendant could point to it.` So we have an id so `aria-activedescendant ` reads it out when you select it

**Comment by melloware** [comment]
oK i will put the aria-label back in as NVDA is clearly not reading this correctly.

**Comment by melloware** [comment]
i will fix this for 15.0.14

**Comment by prusuteodora** [comment]
It seems that only adding the aria-label is not sufficient. There is also needed to be added the attribute: data-item-label in order for the NVDA to read the "MoreText" item.

Here is how in the renderer was modified in order to work on my side:

<img width="928" height="562" alt="Image" src="https://github.com/user-attachments/assets/83e351a3-9a56-4cae-a05d-f1839321f0c7" />

**Comment by melloware** [comment]
@prusuteodora review my PR i made the changes to match yours

**Comment by prusuteodora** [comment]
@melloware looks good! thank you!
